src/api/api_auth_shibuqam.nit

```diff
@@ -59,23 +63,60 @@ class ShibCallback
 		state = state.from_percent_encoding
 		if state != secret then return
 
-		var id = req.string_arg("id")
-		if id == null then return
-		id = id.from_percent_encoding
+		var code = req.string_arg("code")
+		if code == null then return
+		code = code.from_percent_encoding
 
+
+		# Send the code to the oauth server and get something
+		var data = curl_post(code)
+		if data == null then return
+
+		# Try to get something an user for the data
+		var deser_engine = new JsonDeserializer(data)
+		var user = new User.from_deserializer(deser_engine)
+		if deser_engine.errors.not_empty then
+			print_error "Shibuqam OAuth garbage"
+			for e in deser_engine.errors do
+				print_error e
+			end
+			return
+		end
+
+		# Get the player (a new or an old one)
+		var id = user.id
 		var player = config.players.find_by_id(id)
 		if player == null then
 			player = new Player(id)
-			var name = req.string_arg("name")
-			if name != null then name = name.from_percent_encoding
-			player.name = name
-			var avatar = req.string_arg("avatar")
-			if avatar != null then avatar = avatar.from_percent_encoding
-			player.avatar_url = avatar
+			player.name = user.display_name
+			player.avatar_url = user.avatar
 			player.add_achievement(config, new FirstLoginAchievement(player))
 			config.players.save player
 		end
 		session.player = player
 		res.redirect "/player"
 	end
+
+	# Send the token, retrieve information
+	fun curl_post(code: String): nullable String
+	do
+		var url = "https://info.uqam.ca/oauth/info"
+		var request = new CurlHTTPRequest(url)
+		var header = new HeaderMap
+		header["accept"] = "application/json"
+		request.headers = header
+		var post_data = new HeaderMap
+		post_data["code"] = code
+		request.data = post_data
+		var response = request.execute
+		if response isa CurlResponseFailed then
+			print_error "Shibuqam OAuth failed"
```

Comment:
I reused (and improved) the log pattern of https://github.com/nitlang/nit/blob/master/lib/popcorn/pop_auth.nit#L229-L253